cirq-core/cirq/_compat.py

```diff
@@ -71,6 +72,28 @@ def proper_repr(value: Any) -> str:
     return repr(value)
 
 
+def dataclass_repr(value: Any, namespace='cirq'):
+    """Create a Cirq-style repr for a dataclass.
+
+    This creates a repr using the dataclass field names and reprs of their values. It respects
+    the `repr` attribute of dataclass fields if you deign to omit a field from the repr.
+
+    Crucially, this function allows specifying a `namespace` which will be prepended with a "."
+    to the class name.
```

Comment:
Nit: missing args and returns sections.

Reply:
fyi none of the adjacent functions have this. Some of the simpler routines benefit from the inherent clarity of a simple docstring and self-documenting argument names.

Done, though.